ethpm_cli/manifest.py

```diff
@@ -81,45 +86,63 @@ def gen_validate_manifest() -> Optional[Callable[..., Manifest]]:
-        f"{pretty}. \n\n"
-        "Would you like to include all available contract types?"
+        f"{len(all_contract_types)} contract types available.\n\n"
+        f"{pretty}\n"
+        "Would you like to automatically include all available contract types and their sources?"
     )
-    if not flag:
-        raise Exception("Custom contract types are not supported yet.")
-    return build_contract_types(contract_types, solc_output)
 
+    if flag:
+        target_contract_types = all_contract_types
+    else:
+        while True:
+            raw_included_ctypes = input(
+                "Please list the contract types you would like to include, separated by commas: "
+            )
+            target_contract_types = [
+                ct.strip(" ") for ct in raw_included_ctypes.split(",")
+            ]
+            if any(ct not in all_contract_types for ct in target_contract_types):
+                cli_logger.info(f"Invalid contract type selected. Please try again.")
+            else:
+                break
 
-def gen_sources(
-    solc_output: Dict[str, Any], contracts_dir: Path
-) -> Iterable[Callable[..., Manifest]]:
-    available_sources = [
-        str(src.relative_to(contracts_dir)) for src in contracts_dir.glob("**/*.sol")
-    ]
-    contract_types = get_contract_types(solc_output)
-    pretty = "\n".join(sorted(available_sources))
-    flag = parse_bool_flag(
-        "\n"
-        f"{len(available_sources)} sources available.\n\n"
-        f"{pretty}. \n\n"
-        "Would you like to include all available sources?"
-    )
-    if not flag:
-        raise Exception("sorry, we dont support specific source selection yet.")
     inline_source_flag = parse_bool_flag(
-        "Would you like to automatically inline all sources?"
+        "Would you like to inline source files? If not, sources will "
+        "be automatically pinned to IPFS."
+    )
+    target_sources = set(
+        flatten([ct[1] for ct in ctypes_and_sources if ct[0] in target_contract_types])
     )
-    if not inline_source_flag:
-        raise Exception("Sorry, we dont support pinning sources yet.")
-    # todo: validate a source for every contract type
-    return build_sources(contract_types, solc_output, contracts_dir)
+    target_source_names = (Path(src).stem for src in target_sources)
+    generated_contract_types = build_contract_types(target_contract_types, solc_output)
+    if inline_source_flag:
+        generated_sources = build_inline_sources(
+            target_source_names, solc_output, contracts_dir
+        )
+    else:
+        generated_sources = build_pinned_sources(
+            target_source_names, solc_output, contracts_dir
+        )
+    return ((*generated_contract_types), (*generated_sources))
+
+
+@to_list
+def format_contract_types_and_sources_for_display(
+    ctypes_and_sources: Tuple[str]
+) -> Iterable[str]:
+    for ctype in ctypes_and_sources:
+        yield f"{ctype[0]}\n"
+        for src in ctype[1]:
```

Comment:
I think you *likely* have a pending test failure in CI or something due to these values not having their order guaranteed.  To fix this you should probably add a `sorted` call somewhere in here to ensure that the data is output in a deterministic order.